### DISCUSSION TITLE
关于 DnD 大成功和大失败的具体讨论

### DISCUSSION THREAD

**Discussion by kenichiLyon** [sealdice/sealdice-core#997] [Q&A] [answered]
follow up #979 
由于无法从 dicepp 处对齐相关的功能如何加入且存在疑问，特开 discussion 进行讨论。
1、如 #979 所说，`ra`需要加文案，那么`rc`加不加
2、是否有必要做 dnd “出1大失败出20大成功” 的房规？
3、如果做，实现路径可以照抄`ext-coc7.go`的相关部分吗
cc @sealdice/core-team

**Comment by Xiangze-Li** [comment, marked as answer]
严格按照 5e 2014 PHB 来说，只有攻击检定出现 N1 和 N20 是必定失手和暴击命中，属性检定和豁免检定没有对应的规则。

以当前实现方式来说，无法区分属性检定或攻击检定；豁免检定可区分，但实务上往往也不加区分。因此，几乎无法严格按照规则书的方式实现这一功能。

对于房规来说，一个曾见过的房规是将任何检定的 N1 和 N20 都视为必定失败和必定成功，这对于当前实现来说是可行的。但是，**在没有完整实现规则书的前提下实现一种房规** 是否是一个好的实践，是值得商榷的。

具体的代码实现不宜直接确定，这不利于开发。

**Reply by kenichiLyon** [reply]
了解。john佬辛苦

**Reply by YuSheng133** [reply]
是否可以考虑修复[#551](https://github.com/sealdice/sealdice-core/issues/551)后，仿照coc规则中ra指令提供的{$tD100}，返回一个{$tD20}变量，然后把这样的豹语示例写入手册由骰主自行选择添加？
`{%
$tD20 == 20 ? "自然20",
$tD20 == 1 ? " 自然1"
%}`

**Reply by kenichiLyon** [reply]
> 是否可以考虑修复[#551](https://github.com/sealdice/sealdice-core/issues/551)后，仿照coc规则中ra指令提供的{$tD100}，返回一个{$tD20}变量，然后把这样的豹语示例写入手册由骰主自行选择添加？ `{% $tD20 == 20 ? "自然20", $tD20 == 1 ? " 自然1" %}`

这是木落 @fy0 本人在烹饪群提出的观点。可以单独把基础骰的变量抽出。这方面我打算试试看。

**Comment by Szzrain** [comment]
于 #1265 已经解决
